### DISCUSSION TITLE
Is Vulkan renderer removed?

### DISCUSSION THREAD

**Discussion by ArinoJenynof** [86Box/86Box#3361] [Q&A] [answered]
It seems that i can't select vulkan renderer anymore, it's just gone from the menu. Happened when i update from build 4427 to 4428. The binaries were from jenkins.

**Before ([commit: 64ad9e5](https://github.com/86Box/86Box/commit/64ad9e5fbaea9b218c5c165397673dee18614a3c))**:
![4427](https://github.com/86Box/86Box/assets/36990619/fb4d5d91-76cd-4efa-bf15-2a3fc10994c8)

**After ([commit: 22c0fd2](https://github.com/86Box/86Box/commit/22c0fd29a360f8dde4180e744038e5989e2e29d2))**:
![4428](https://github.com/86Box/86Box/assets/36990619/8ee9df20-48b7-4c88-919c-425425d91512)

**Comment by parkerlreed** [comment]
What about on current builds? 4428 is relatively old. 4631 is the current.

**Reply by parkerlreed** [reply]
For what it's worth even 4427 on Linux fails to load Vulkan (Along with current releases)

```
Failed to create Vulkan instance: -9
Failed to create platform Vulkan instance
```

**Reply by parkerlreed** [reply]
Browsing the Discord it looks like Vulkan was only ever on Windows and it got disabled in a QT5 static cleanup around the time you mentioned.

**Reply by parkerlreed** [reply]
I can confirm for Linux at least, building 86Box from source and linking against system Qt enables Vulkan. 

Not sure for Windows, sorry.

**Reply by ArinoJenynof** [reply]
> What about on current builds? 4428 is relatively old. 4631 is the current.

Yes, it's gone from 4428 onwards. I was updating to 4631 from a very old version, i forgot which, to find out that GPU activity icon in the tray doesn't light up. Then i see vulkan is gone and it was selecting Qt (Software). I bisect it to that commit.



> Browsing the Discord it looks like Vulkan was only ever on Windows and it got disabled in a QT5 static cleanup around the time you mentioned.

Ah, so perhaps it was intentional? maybe it was never ready.



> I can confirm for Linux at least, building 86Box from source and linking against system Qt enables Vulkan.
> 
> Not sure for Windows, sorry.

Thanks for the info, i guess i'm going to try to compile from source. Will report back if i can 👍

**Reply by parkerlreed** [reply, marked as answer]
Found the offending commit https://github.com/msys2/MINGW-packages/commit/f6e5c742915df7b0624f89305c58a550b5b27d4a#diff-43630faca9601f37d00795e45d987346c3db25b7e9a9792be78bc535a1029879L16

If you can revert to the commit beforehand for QT5 static and redo an 86Box build from master that might work.

**Reply by ArinoJenynof** [reply]
> Found the offending commit [msys2/MINGW-packages@f6e5c74#diff-43630faca9601f37d00795e45d987346c3db25b7e9a9792be78bc535a1029879L16](https://github.com/msys2/MINGW-packages/commit/f6e5c742915df7b0624f89305c58a550b5b27d4a#diff-43630faca9601f37d00795e45d987346c3db25b7e9a9792be78bc535a1029879L16)
> 
> If you can revert to the commit beforehand for QT5 static and redo an 86Box build from master that might work.

Wanted to reply right before your answer came in. I don't know how to revert to that commit though. I might know MSYS2, but i don't know that much to know how to install some package from some commit.

A lucky google search landed me in stackoverflow post of how to check vulkan support in Qt. Looks like `qt5-static` package in MSYS2 wasn't build with vulkan support (`#define QT_FEATURE_vulkan -1` in `qtgui-config.h`), but the non-static `qt5` do support vulkan. Might be related to that commit but i don't know which line caused that. So i built the optimised build and...

![4631](https://github.com/86Box/86Box/assets/36990619/e80a2011-da6f-446d-8842-e2b0a4777326)

Furthermore, looks like `qt6-static` is built with vulkan support. But when i tried to compile with qt6 (`USE_QT6` when generating i believe), it failed linking to, uhh, something. Had a lot of noise so i guess qt6 is not meant to be used yet.

So that's it. Thanks for the help, appreciate it 😁👍

**Reply by parkerlreed** [reply]
Good news!!

https://github.com/msys2/MINGW-packages/pull/17186

So hopefully within the next few nightly builds this will be included.

**Reply by parkerlreed** [reply]
It's live!

https://ci.86box.net/job/86Box/4632/

**Reply by ArinoJenynof** [reply]
Good news indeed. Again, thank you for that quick handling with upstream 😁
